markets/retrievaladapter/client.go

```diff
@@ -56,7 +59,10 @@ func (rcn *retrievalClientNode) CreatePaymentVoucher(ctx context.Context, paymen
 	if err != nil {
 		return nil, err
 	}
-	return voucher, nil
+	if voucher.Voucher == nil {
```

Comment:
@hannahhoward in order to change the return value here we would first need to update the `RetrievalClientNode ` interface in retrieval markets. So for now I'm just returning an error here if the funds are insufficient, let me know if that makes sense to you

Reply:
that's fine